treeherder/model/models.py

```diff
@@ -333,6 +334,32 @@ def search(self, search_term):
         return {"open_recent": open_recent, "all_others": all_others}
 
 
+class BugzillaComponent(models.Model):
+    product = models.CharField(max_length=60)
+    component = models.CharField(max_length=60)
+
+    class Meta:
+        db_table = 'bugzilla_component'
+        verbose_name_plural = 'bugzilla_components'
+        unique_together = ("product", "component")
+
+    def __str__(self):
+        return "{0} :: {1}".format(self.product, self.component)
+
+
+class FilesBugzillaMap(models.Model):
+    path = models.CharField(max_length=255, unique=True, db_index=True)
+    file_name = models.CharField(max_length=255, db_index=True)
+    bugzilla_component = models.ForeignKey('BugzillaComponent', on_delete=models.CASCADE)
+
+    class Meta:
+        db_table = 'file_bugzilla_component'
+        verbose_name_plural = 'files_bugzilla_components'
+
+    def __str__(self):
+        return "{0}".format(self.path)
```

Comment:
How big do you think these tables will get and will there ever be a need to expire old data?

Reply:
A current run creates 326k rows. When mozilla-esr78 switches to inactive in ~3 months, it will drop to 292k. The expiration is part of the ingestion script (`paths_removed`). The manual action needed is to remove the `life_cycle_order` attribute from `repository.json` when ESR trees reach their EOL.